### PR TITLE
[MFTF] Refactoring of Search actions on Storefront

### LINKED ISSUE #31251: [Issue] [MFTF] Refactoring of Search actions on Storefront
This issue is automatically created based on existing pull request: magento/magento2#31240: [MFTF] Refactoring of Search actions on Storefront

---------

### Description (*)
- **_StorefrontAssertProductNameIsNotOnProductMainPageActionGroup_** has been added
- Tests have been refactored with existing **_StorefrontCheckQuickSearchStringActionGroup,_** **_StorefrontAssertProductNameOnProductMainPageActionGroup_** and newly created **_StorefrontAssertProductNameIsNotOnProductMainPageActionGroup_**
**Please, note:** **_AdminSubmitCategoriesPopupActionGroup_** (not related to search) has been added to some tests to prevent tests failure

### PR DESCRIPTION
**Pull request by AnnaAPak**
### Description (*)
- **_StorefrontAssertProductNameIsNotOnProductMainPageActionGroup_** has been added
- Tests have been refactored with existing **_StorefrontCheckQuickSearchStringActionGroup,_** **_StorefrontAssertProductNameOnProductMainPageActionGroup_** and newly created **_StorefrontAssertProductNameIsNotOnProductMainPageActionGroup_**
**Please, note:** **_AdminSubmitCategoriesPopupActionGroup_** (not related to search) has been added to some tests to prevent tests failure

### Resolved issues:
1. [x] resolves magento/magento2#31251: [MFTF] Refactoring of Search actions on Storefront

### CODE REVIEW

**Comment by AnnaAPak** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Comment by sidolov** [comment]
@magento create issue

File: app/code/Magento/Catalog/Test/Mftf/ActionGroup/StorefrontAssertProductNameIsNotOnProductMainPageActionGroup.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="StorefrontAssertProductNameIsNotOnProductMainPageActionGroup">
```
Comment (rogyar) [inline app/code/Magento/Catalog/Test/Mftf/ActionGroup/StorefrontAssertProductNameIsNotOnProductMainPageActionGroup.xml]: Please, rename the action group according to the correct naming conventions

**Review by rogyar — CHANGES_REQUESTED** [review]

File: app/code/Magento/Catalog/Test/Mftf/ActionGroup/StorefrontAssertProductNameIsNotOnProductMainPageActionGroup.xml (hunk 1 above)
Comment (AnnaAPak) [reply app/code/Magento/Catalog/Test/Mftf/ActionGroup/StorefrontAssertProductNameIsNotOnProductMainPageActionGroup.xml]: I have addressed this. Thank you!

**Review by rogyar — APPROVED** [review]

**Comment by engcom-Delta** [comment]
**Note:** QA not applicable

**Comment by engcom-Delta** [comment]
**Note:** Functional Tests CE is failed

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-8562](https://jira.corp.magento.com/browse/ENGCOM-8562) has been created to process this Pull Request

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
@magento run all tests

**Pull request merged**